builder_test.go

```diff
@@ -72,70 +72,70 @@ func Test_Builder(t *testing.T) {
 		},
 		{
 			"complex message, text with special chars",
-			NewBuilder().Local("$var", Literal("greeting")),
+			NewBuilder().Local("$var", Expr("greeting")),
 			".local $var = { greeting }\n{{}}",
 		},
 		{
 			"complex message, local declaration",
 			NewBuilder().
-				Local("$hostName", Var("$host")).
-				Expr(Var("$hostName")),
+				Local("$hostName", Expr(Var("host"))).
+				Expr(Expr(Var("hostName"))),
 			".local $hostName = { $host }\n{{{ $hostName }}}",
 		},
 		{
 			"complex message, input declaration",
 			NewBuilder().
-				Input(Var("$host")).
-				Expr(Var("$host")),
+				Input(Expr(Var("host"))).
+				Expr(Expr(Var("host"))),
 			".input { $host }\n{{{ $host }}}",
 		},
 		{
 			"complex message, input and local declaration",
 			NewBuilder().
-				Local("$hostName", Var("$host")).
-				Input(Var("$host")).
-				Expr(Var("$host")),
+				Local("$hostName", Expr(Var("host"))).
+				Input(Expr(Var("host"))).
+				Expr(Expr(Var("host"))),
 			".input { $host }\n.local $hostName = { $host }\n{{{ $host }}}",
 		},
 		{
 			"complex message, matcher with multiple keys",
 			NewBuilder().
 				Match(
-					Var("$i"),
-					Var("$j"),
+					Expr(Var("i")),
+					Expr().Var("j"),
 				).
 				Keys(1, 2).Text("{first}").
-				Keys(2, 0).Text("second ").Expr(Var("$i")).
-				Keys(3, 0).Expr(Literal("\\a|")).
-				Keys("*", "*").Expr(Literal(1)),
+				Keys(2, 0).Text("second ").Expr(Expr().Var("i")).
+				Keys(3, 0).Expr(Expr("\\a|")).
+				Keys("*", "*").Expr(Expr(1)),
 			".match { $i } { $j }\n1 2 {{\\{first\\}}}\n2 0 {{second { $i }}}\n3 0 {{{ |\\\\a\\|| }}}\n* * {{{ 1 }}}",
 		},
 		{
 			"complex message, matcher with multiple keys and local declarations",
 			NewBuilder().
-				Input(Var("$i")).
-				Local("$hostName", Var("$i")).
+				Input(Expr().Var("i")).
+				Local("$hostName", Expr().Var("i")).
 				Match(
-					Var("$i"),
-					Var("$j"),
+					Expr().Var("i"),
+					Expr().Var("j"),
 				).
 				Keys(1, 2).Text("{first}").
-				Keys(2, 0).Text("second ").Expr(Var("$i")).
-				Keys(3, 0).Expr(Literal("\\a|")).
-				Keys("*", "*").Expr(Literal(1)),
+				Keys(2, 0).Text("second ").Expr(Expr().Var("i")).
+				Keys(3, 0).Expr(Expr("\\a|")).
+				Keys("*", "*").Expr(Expr(1)),
 			".input { $i }\n.local $hostName = { $i }\n.match { $i } { $j }\n1 2 {{\\{first\\}}}\n2 0 {{second { $i }}}\n3 0 {{{ |\\\\a\\|| }}}\n* * {{{ 1 }}}",
 		},
 		{
 			"spacing",
 			NewBuilder().
 				Match(
-					Var("$i"),
-					Var("$j"),
+					Expr().Var("i"),
+					Expr().Var("j"),
 				).
 				Keys(1, 2).Text("{first}").
-				Keys(2, 0).Text("second ").Expr(Var("$i")).
-				Keys(3, 0).Expr(Literal("\\a|")).
-				Keys("*", "*").Expr(Literal(1)).
+				Keys(2, 0).Text("second ").Expr(Expr().Var("i")).
```

Comment:
Hmm `Expr(Expr().Var("i")).` this doesn't look too good.

Reply:
```go
// it is already possible, both ways are correct
Expr(Var("i")
Expr().Var("i")

// literal - no need to specify type, literal implicitly
Expr("Hello")
Expr(Literal("Hello"))
Expr().Literal("Hello")
```